File: PilluleBoxApp/PilluleBox/app/src/main/java/Models/ScheduleModes/SequentialMode.java
package Models.ScheduleModes;

import java.io.Serializable;
import java.sql.Time;
import java.sql.Date;

public class SequentialMode implements Serializable {
    private int id;
    private String medicineName;
    private Date startDate;
    private Date endDate;
    private Time period;
    private int limitTimesConsumption;
    private boolean affectedPeriods;
    private int currentTimesConsumption;

    public SequentialMode(String medicineName, Date startDate,
                          Date endDate, Time period, int limitTimesConsumption,
                          boolean affectedPeriods) {
        this.medicineName = medicineName;
        this.startDate = startDate;
        this.endDate = endDate;
        this.period = period;
        this.limitTimesConsumption = limitTimesConsumption;
        this.affectedPeriods = affectedPeriods;
        currentTimesConsumption = 0;
    }

    public SequentialMode(int id, String medicineName, Date startDate,
                          Date endDate, Time period, int limitTimesConsumption,
                          boolean affectedPeriods, int currentTimesConsumption) {
        this.id = id;
        this.medicineName = medicineName;
        this.startDate = startDate;
        this.endDate = endDate;
        this.period = period;
        this.limitTimesConsumption = limitTimesConsumption;
        this.affectedPeriods = affectedPeriods;
        this.currentTimesConsumption = currentTimesConsumption;
    }

    // Getters y Setters actualizados
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getMedicineName() { return medicineName; }
    public void setMedicineName(String medicineName) { this.medicineName = medicineName; }

    public Date getStartDate() { return startDate; }
    public void setStartDate(Date startDate) { this.startDate = startDate; }

    public Date getEndDate() { return endDate; }
    public void setEndDate(Date endDate) { this.endDate = endDate; }

    public Time getPeriod() { return period; }
    public void setPeriod(Time period) { this.period = period; }

    public int getLimitTimesConsumption() { return limitTimesConsumption; }
    public void setLimitTimesConsumption(int limitTimesConsumption) { this.limitTimesConsumption = limitTimesConsumption; }

    public boolean isAffectedPeriods() { return affectedPeriods; }
    public void setAffectedPeriods(boolean affectedPeriods) { this.affectedPeriods = affectedPeriods; }

    public int getCurrentTimesConsumption() { return currentTimesConsumption; }
    public void setCurrentTimesConsumption(int currentTimesConsumption) { this.currentTimesConsumption = currentTimesConsumption; }
}

File: PilluleBoxESP32/PilluleBox/SequentialMode.cpp
#include "pillulebox.h"

SequentialMode::SequentialMode(int id) {
  this->id = id;
}

SequentialMode::~SequentialMode() {}

int SequentialMode::getId() const {
  return id;
}

const char* SequentialMode::getMedicineName() const {
  return medicine_name;
}

time_t SequentialMode::getStartDate() const {
  return start_date;
}

time_t SequentialMode::getEndDate() const {
  return end_date;
}

tm SequentialMode::getPeriod() const {
  return period;
}

uint8_t SequentialMode::getLimitTimesConsumption() const {
  return limit_times_consumption;
}

bool SequentialMode::getAffectedPeriods() const {
  return affected_periods;
}

uint8_t SequentialMode::getCurrentTimesConsumption() const {
  return current_times_consumption;
}

void SequentialMode::setId(int id) {
  this->id = id;
}

void SequentialMode::setMedicineName(const char* name) {
  strncpy(medicine_name, name, sizeof(medicine_name) - 1);
  medicine_name[sizeof(medicine_name) - 1] = '\0';  // Asegura la terminación nula
}

void SequentialMode::setStartDate(time_t date) {
  start_date = date;
}

void SequentialMode::setEndDate(time_t date) {
  end_date = date;
}

void SequentialMode::setPeriod(const tm& period) {
  this->period = period;
}

void SequentialMode::setLimitTimesConsumption(uint8_t limit) {
  limit_times_consumption = limit;
}

void SequentialMode::setAffectedPeriods(bool affected) {
  affected_periods = affected;
}

void SequentialMode::setCurrentTimesConsumption(uint8_t times) {
  current_times_consumption = times;
}